client/state/ui/reducer.js

```diff
@@ -91,13 +92,18 @@ export const isNotificationsOpen = function( state = false, { type } ) {
 	return state;
 };
 
+export const hideMasterbar = createReducer( false, {
+	[ MASTERBAR_HIDE ]: ( state, { hide } ) => hide,
+} );
```

Comment:
how about we remove some abstraction and overhead here since it's a simple reducer?

```js
export const hideMasterbar = ( state = false, { type, hide } ) =>
	type === MASTERBAR_HIDE
		? hide
		: state;
```

Also, is this the best way to represent it? I could see it making sense to store a property like `masterbarVisibility` instead of the action-nuance of `hide` or `show`